### DISCUSSION TITLE
DIA_Lothar_HelloAgain

### DISCUSSION THREAD

**Discussion by Kor-Angar** [dosinabox/g2nr_unofficial_update#472] [Ideas]
у Лотара стоит проверка на официальный проход в ратушу

```
func int DIA_Lothar_HelloAgain_Condition()
{
	if(Mil_305_schonmalreingelassen == TRUE)
	{
		return TRUE;
	};
};
```

таким образом если герой проник вверхний квартал неофициально (_например заморозил паладинов и прошёл или спровоцировал их на атаку и пробежал или ещё как то_), то этого диалога с Лотаром не будет и Лотар начнёт посылать героя к Андре :-)

соответственно можно сделать проверку у Лотара на несанкционнированный доступ вверхний квартал if(Mil_305_schonmalreingelassen == false) и в таком случае что то делать, например кричать "Аларм" и  нападать на героя :-)

**Comment by dosinabox** [comment]
Возможно, нужна переменная для отслеживания попадания ГГ в верхний квартал, по аналогии с PlayerEnteredCity, и расширенное назначение (диалоги с жителями, ополчением, паладинами, пересечения триггеров, нажатия секретных кнопок). Можно подумать, где её ещё можно применить. Подозреваю, что Mil_305_schonmalreingelassen использует для проверки не только Лотар.

> в таком случае что то делать, например кричать "Аларм" и нападать на героя

Такое же поведение можно будет назначить и стражникам входа в верхний квартал.

**Comment by Kor-Angar** [comment]
> Возможно, нужна переменная для отслеживания попадания ГГ в верхний квартал, по аналогии с PlayerEnteredCity

было бы замечательное

**Comment by Kor-Angar** [comment]
а ещё можно добавить переменную Bennet_PRISON ?
я пока реализовал посадку Беннета в 3 главе, нашел несколько диалогов где в оригинальном коде, где некорректно определялся сидит или нет Беннет :-)

**Comment by dosinabox** [comment]
(Kapitel < 3) - Беннет на ферме
((Kapitel == 3) && (MIS_RescueBennet != LOG_SUCCESS)) - Беннет в тюрьме
(MIS_RescueBennet == LOG_SUCCESS) - Беннет вышел из тюрьмы

Такая логика подходит без ошибок на оригинальное развитие событий, а для твоих правок уже нужна отдельная переменная. Но только для твоих, а так я не буду вводить новую переменную. Я наоборот стараюсь избавляться от ненужных или неиспользуемых.

**Comment by dosinabox** [comment]
> в таком случае что то делать, например кричать "Аларм" и нападать на героя

А если ГГ уже маг?)

**Reply by Kor-Angar** [reply]
только для нейтрала и наёмника :-)

**Comment by dosinabox** [comment]
Готово. Только без аларма)

**Comment by Kor-Angar** [comment]
а скинь фрагмент кода или ссылку на правку. а то сейчас уже не так понятно про что речь :-)

**Comment by Kor-Angar** [comment]
> Возможно, нужна переменная для отслеживания попадания ГГ в верхний квартал, по аналогии с PlayerEnteredCity, и расширенное назначение (диалоги с жителями, ополчением, паладинами, пересечения триггеров, нажатия секретных кнопок). 

в коде по моему такая переменная есть.  **PlayerEnteredUpperCity** не оно?

**Comment by dosinabox** [comment]
> а скинь фрагмент кода или ссылку на правку. а то сейчас уже не так понятно про что речь :-)

https://github.com/dosinabox/g2nr_unofficial_update/commit/ec43e31dae993755c53798504ff64d3a56f0d8b5#diff-a05ff3733cacfa2c8a6df4f1a744fb0ee5f47fecb9b111ed1696da1f314c6d8f
В частности, DIA_Lothar_HelloAgain_Condition().

> в коде по моему такая переменная есть. **PlayerEnteredUpperCity** не оно?

Да, это моя переменная, но сейчас она не используется и не назначается. Пока для нее нет применения.

**Comment by Kor-Angar** [comment]
> а ещё можно добавить переменную Bennet_PRISON ?

можно использовать `func int` что бы голова не болела
```D
func int C_Bennet_PRISON()
{
    return Bennet_PRISON;
};
```
в самой функции можно делать разные проверки. например то что писала выше Дося

> (Kapitel < 3) - Беннет на ферме
> ((Kapitel == 3) && (MIS_RescueBennet != LOG_SUCCESS)) - Беннет в тюрьме
> (MIS_RescueBennet == LOG_SUCCESS) - Беннет вышел из тюрьмы

или проверка на то где находится беннет `Npc_GetDistToWP`

**Reply by Kor-Angar** [reply]
```D
func int C_Bennet_PRISON()
{
    if (MIS_RescueBennet == LOG_SUCCESS)
    {
        return false;
    };

    if Hlp_IsValidNpc(SLD_809_Bennet_Prison)
    {
        return true;
    };
    return false;
    //return Bennet_PRISON;
};
```
